### ISSUE TITLE
 Internal error NFUnitCheck.checkUnits: unit check module failed

### ISSUE THREAD

**Issue by bilderbuchi** [OpenModelica/OpenModelica#10057]
### Description

In some models, I have recently observed the unit check module failing with an internal error when pressing "Check model" in OMEdit.
I have whittled down one model to this minimized reproducer:
```modelica
model UnitCheckFailingVolume "Minimal test case for erroring unit testing"
  //this triggers: Internal error NFUnitCheck.checkUnits: unit check module failed
  Modelica.Fluid.Machines.SweptVolume sweptVolume(nPorts = 1);
  // nPorts=0 (the default) does not trigger the unit checking error
  inner Modelica.Fluid.System system;
end UnitCheckFailingVolume;
```
(The  `system` is just there to avoid distracting translation warnings about a missing system, it's not necessary to trigger the problem.

### Steps to Reproduce

Load the model above. Press "Check model". Observe in the log

>Translation Error
[C:/OM120/OM64bit/OMCompiler/Compiler/NFFrontEnd/NFUnitCheck.mo: 83:5-83:91]: Internal error NFUnitCheck.checkUnits: unit check module failed

### Expected Behavior
The unit checking module not crashing, and reporting what the problem is, if any.
If you have the default value `nPorts = 0` part, the error disappears.

### Version and OS
 - OpenModelica Version: 1.20
 - OS: Windows 10, 64 bit
 - In Options-Simulation-Additional Translation Settings, I have `--unitChecking`, if that's relevant.

**Comment by casella** [comment]
In the past array variables were not subject to unit checking, see #5685. This was fixed four months ago in #9446, maybe there is still something wrong with it.

@perost can you please check?

**Comment by bilderbuchi** [comment]
Thanks for the quick fix, confirmed!
